File: python/swap.py
def swap1(x, y):
	temp=y
	y=x
	x=temp
	return x, y

def swap2(x, y):
	z=x+y
	y=z-y
	x=z-x
	return x, y



# main
x=2
y=3
x,y = swap1(x, y)
print("swap1")
print(f"x is {x}")
print(f"y is {y}")

x=2
y=3
x,y = swap2(x, y)
print("swap1")
print(f"x is {x}")
print(f"y is {y}")

x=2
y=3
x,y = y,x
print("swap3")
print(f"x is {x}")
print(f"y is {y}")

File: cpp/swap.cpp
#include <iostream>
using namespace std;

void swap1(int x,int y)
{
	int temp = y;
	y = x;
	x = temp;
 }
 void swap2(int x,int y)
 {
	int z=x+y;
	y=z-y;
	x=z-x;
 }	
int main()
{
	int x =2;
	int y =3;

	swap1(x,y);
	cout<<"x is "<< x <<"\n";
    cout<<"y is "<< y <<"\n";


	x =2;
	y =3;
	swap2(x,y);
	cout<<"x is "<< x <<"\n";
    cout<<"y is "<< y <<"\n";

	return 0;
}